services/cloudsecuritygroup/securitygroupazure/securitygroupazure.go

```diff
@@ -0,0 +1,119 @@
+	ResourceGroup     string         `json:"resourceGroup"`
+	CloudAccountID    string         `json:"cloudAccountId"`
+	Tags              []Tags         `json:"tags"`
+	InboundServices   []BoundService `json:"inboundServices"`
+	OutboundServices  []BoundService `json:"outboundServices"`
+}
+
+type CloudSecurityGroupResponse struct {
+	ID                      string         `json:"id"`
+	ExternalSecurityGroupID string         `json:"externalSecurityGroupId"`
+	AccountID               int            `json:"accountId"`
+	CloudAccountName        string         `json:"cloudAccountName"`
+	LastUpdatedByDome9      bool           `json:"lastUpdatedByDome9"`
+	Error                   Error          `json:"error"`
+	CloudAccountID          string         `json:"cloudAccountId"`
+	Name                    string         `json:"name"`
+	Description             string         `json:"description"`
+	Region                  string         `json:"region"`
+	ResourceGroup           string         `json:"resourceGroup"`
+	InboundServices         []BoundService `json:"inboundServices"`
+	OutboundServices        []BoundService `json:"outboundServices"`
+	IsTamperProtected       bool           `json:"isTamperProtected"`
+	Tags                    []Tags         `json:"tags"`
+}
+
+type Tags struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type Scope struct {
+	Type string            `json:"type"`
+	Data map[string]string `json:"data"`
+}
+
+type BoundService struct {
+	Direction             string   `json:"direction"`
+	Name                  string   `json:"name"`
+	Description           string   `json:"description"`
+	Priority              int      `json:"priority"`
+	Access                string   `json:"access"`
+	Protocol              string   `json:"protocol"`
+	SourcePortRanges      []string `json:"sourcePortRanges"`
+	SourceScopes          []Scope  `json:"sourceScopes"`
+	DestinationPortRanges []string `json:"destinationPortRanges"`
+	DestinationScopes     []Scope  `json:"destinationScopes"`
+	IsDefault             bool     `json:"isDefault"`
+}
+
+type Error struct {
+	Action       string `json:"action"`
+	ErrorMessage string `json:"errorMessage"`
+}
+
+func (service *Service) Get(id string) (*CloudSecurityGroupResponse, *http.Response, error) {
+	v := new(CloudSecurityGroupResponse)
+	relativeURL := fmt.Sprintf("%s/%s", cloudsecuritygroup.RESTfulPathAzureSecurityGroup, id)
+	resp, err := service.Client.NewRequestDo("GET", relativeURL, nil, nil, v)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return v, resp, nil
+}
+
+func (service *Service) GetAll() (*[]CloudSecurityGroupResponse, *http.Response, error) {
+	v := new([]CloudSecurityGroupResponse)
+	resp, err := service.Client.NewRequestDo("GET", cloudsecuritygroup.RESTfulPathAWS, nil, nil, v)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return v, resp, nil
+}
+
+func (service *Service) Create(body *CloudSecurityGroupRequest) (*CloudSecurityGroupResponse, *http.Response, error) {
```

Comment:
Pass `CloudSecurityGroupRequest` by value and let the compiler decider to pass it by pointer or by value.